dev/integration_tests/ios_add2app_life_cycle/ios_add2app/AppDelegate.m

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "AppDelegate.h"
+#import "MainViewController.h"
+
+@interface AppDelegate ()
+
+@end
+
+@implementation AppDelegate { }
+
+- (BOOL)application:(UIApplication *)application
+    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
+
+  MainViewController *mainViewController = [[MainViewController alloc] init];
+  UINavigationController *navigationController = [[UINavigationController alloc]
+      initWithRootViewController:mainViewController];
+
+  navigationController.navigationBar.translucent = NO;
+
+  _engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
```

Comment:
Use property getters setters, not the backing instance variables.  A good rule of thumb is to only set the instance variables in -init or -dealloc (when `self` shouldn't be referenced) or in an overridden setter.
1. Properties are expected to be key-value observable, but setting the backing instance variable will not trigger notifications.
1. Subclasses may override the property setter (maybe the subclass has some setting side-effect, or it's not using this instance variable at all), and setting the backing instance variable would bypasses it.
1. Likewise subclasses may override the property getter and not use this instance variable.

```objective-c
self.engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
[self.engine runWithEntrypoint:nil];
```